CONTRIBUTING.md

```diff
@@ -0,0 +1,63 @@
+4. Send a pull request (PR) and work with us until it gets merged and published. Contributions may need some modifications, so  a few rounds of review and fixing may be necessary. We will include your name in the ChangeLog :)
+5. For quick merging, the contribution should be short, and concentrated on a single feature or topic. The larger the contribution is, the longer it would take to review it and merge it.
+6. mbed TLS is release with Apache license, and as such, all the added files should include the Apache license header.
+
+### Backports
+
+mbed TLS maintains some legacy branches, which are released as LTS versions. mbed TLS should follow backwards compatibility rules, to fit with existing users. As such, backporting to these branches should be handled according to the following rules:
+  
+1. If the contribution is a new feature or enhancement, no backporting is needed.
+2. Bug fixes should be backported to the legacy branches containing these bugs.
+3. Changes in the API do not require backporting. If a bug fix introduced a new API, such as new error codes, the bug fix should be implemented differently in the legacy branch.
+
+It would be highly appreciated if a contribution would be backported to a legacy branch as well.  
+At the moment, the legacy branches are:
+  
+1. [mbedtls-1.3](https://github.com/ARMmbed/mbedtls/tree/mbedtls-1.3)  
+2. [mbedtls-2.1](https://github.com/ARMmbed/mbedtls/tree/mbedtls-2.1)  
+3. [development](https://github.com/ARMmbed/mbedtls/tree/development)  
+
+### Tests
+
+As mentioned, tests that show the correctness of the feature or bug fix should be added to the Pull Request, if no such tests exist.  
+mbed TLS includes an elaborate test suite in `tests/` that initially requires Perl to generate the tests files (e.g. `test_suite_mpi.c`). These files are generated from a `function file` (e.g. `suites/test_suite_mpi.function`) and a `data file` (e.g. `suites/test_suite_mpi.data`). The function file contains the test functions. The data file contains the test cases, specified as parameters that will be passed to the test function.
+
+Sample applications, if needed, should be modified as well.
+
+### Continuous Integration Tests
+
+Once a PR has been made, the Continuous Integration (CI) tests are triggered and run. You should follow the result of the CI tests, and fix failures. 
+It is advised to enable the [githooks scripts](https://github.com/ARMmbed/mbedtls/tree/development/tests/git-scripts) prior to pushing your changes, for catching some of the issues as early as possible.
+
+### Documentation
+
+mbed TLS should be well documented. If documentation is needed, speak out!
+
+1. All interfaces should be documented through Doxygen. New APIs should introduce Doxygen documentation.
+2. Complex parts in the code should include comments.
+3. If needed, a Readme file is advised
+4. If a KB article should be added, write this as a comment in the PR description.
```

Comment:
It is not clear what KB is... Perhaps this should write `knowledgebase (KB)` and be a link to the tls.mbed.org website